core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/SolrHelperServiceImpl.java

```diff
@@ -427,7 +434,14 @@ public Object getPropertyValue(Object object, Field field) throws NoSuchMethodEx
     @Override
     public Object getPropertyValue(Object object, String propertyName) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         String[] components = propertyName.split("\\.");
-        return getPropertyValueInternal(object, components, 0);
+        Object propertyValueInternal = getPropertyValueInternal(object, components, 0);
+        if(propertyValueInternal instanceof String){
+            String enabled = environment.getProperty("exploitProtection.xssEnabled", "false");
+            if(Boolean.parseBoolean(enabled)){
+                return StringEscapeUtils.unescapeHtml((String) propertyValueInternal);
```

Comment:
Why do we check this property here but not in the other changes in `FormBuilderServiceImpl`?

Reply:
@cja769 I think solr helper I was fixing representation from search/search capabilities on site.
SolrHelper is used in site, and maybe indexing, and FormBuilder is used in admin. So this is completely different places and they don't share anything common